src/lib/journey/journey.interfaces.ts

```diff
@@ -2,20 +2,52 @@ import type { FRStep, FRLoginFailure, FRLoginSuccess } from '@forgerock/javascri
 import type { Step, StepOptions } from '@forgerock/javascript-sdk/lib/auth/interfaces';
 import type { Writable } from 'svelte/store';
 
+import type { Maybe } from '$lib/interfaces';
+
+// type Callbacks = CallbackType.BooleanAttributeInputCallback
+// | CallbackType.ChoiceCallback
+// | CallbackType.ConfirmationCallback
+// | CallbackType.DeviceProfileCallback
+// | CallbackType.HiddenValueCallback
+// | CallbackType.KbaCreateCallback
+// | CallbackType.MetadataCallback
+// | CallbackType.NameCallback
+// | CallbackType.NumberAttributeInputCallback
+// | CallbackType.PasswordCallback
+// | CallbackType.PollingWaitCallback
+// | CallbackType.ReCaptchaCallback
+// | CallbackType.RedirectCallback
+// | CallbackType.SelectIdPCallback
+// | CallbackType.StringAttributeInputCallback
+// | CallbackType.SuspendedTextOutputCallback
+// | CallbackType.TermsAndConditionsCallback
+// | CallbackType.TextOutputCallback
+// | CallbackType.ValidatedCreatePasswordCallback
+// | CallbackType.ValidatedCreateUsernameCallback;
+
 export interface JourneyStore extends Pick<Writable<JourneyStoreValue>, 'subscribe'> {
   next: (prevStep?: StepTypes, nextOptions?: StepOptions) => void;
   reset: () => void;
 }
 export interface JourneyStoreValue {
   completed: boolean;
-  error: {
-    code: number | null;
-    message: string | null;
-    step: Step | undefined;
-  } | null;
+  error: Maybe<{
+    code: Maybe<number>;
+    message: Maybe<string>;
+    step: Maybe<Step>;
+  }>;
   loading: boolean;
   step: StepTypes;
   successful: boolean;
-  response: Step | null | undefined;
+  response: Maybe<Step>;
+}
+export type StepTypes = WidgetStep | FRLoginSuccess | FRLoginFailure | null;
+export interface WidgetStep extends FRStep {
+  /**
+   * TODO: It's complicated, but this `any` is needed for `callback` compatibility with
+   * the `mapCallbackToComponent`. FRStep types this callback array as `FRCallback[]`
+   * which is not _exactly_ correct as it's an array of various types of callbacks, all
+   * which extend FRCallback. So, there are heterogeneous types within the array.
+   */
+  callbacks: any[];
```

Comment:
To get a perfectly clean `npm run check`, I had to override the `callbacks` property on `FRStep` type with an `any[]`. the original type is `FRCallback[]`, but there's not enough overlap between all the actual callbacks, like `ChoiceCallback`, `NameCallback`, `KBACallback` for the callback components.

The issue is at the exact place the type error occurs, you can't type the value since it's in the Svelte template portion of the code :(